### ISSUE TITLE
Control history size

### ISSUE THREAD

**Issue by eyvindbjartiandersen** [DaveDavenport/rofi#613]
I would be cool to be able to control history size. For me personally the history limit is too low. Maybe the ability to specify favorite programs could be useful.

**Comment by DaveDavenport** [comment]
I don't understand the second part. If not directly related to the history size, please keep in a separate issue. (History is based on usage, so favorite programs should show up automatically)

**Comment by ponderstibbons** [comment]
I'd also be interested controlling the history size. From skimming the source, it seems to be an easy change. 

* add `history_size` to the settings struct in settings.h
* add it to the xrmOptions struct so it can be parsed
* `use settings->history_size` instead of `HISTORY_MAX_ENTRIES`

Did I miss anything? If not and you'd be interested in merging, I could write a patch on the weekend.

**Comment by DaveDavenport** [comment]
We currently are in a feature freeze for the 1.4 release. 
I am a bit hesitant in allowing larger history sizes, because with it current implementation it is a perfect way to slow down rofi.
The way history works should probably be re-done.

**Comment by ponderstibbons** [comment]
Ok, didn't know that. Imo it's not so critical to warrant a rewrite, though, at least for my use case.

**Comment by DaveDavenport** [comment]
Ill leave it open as an enhancement for the future.

**Comment by ponderstibbons** [comment]
Thanks. I probably won't have time to contribute to a bigger rewrite, though. So for everybody who can't wait and is content with a quick hack I uploaded a gist [here](https://gist.github.com/ponderstibbons/e411a7b34fc3d36ee570d5fa5efcfdfb) that just changes the global define.

**Comment by pmrv** [comment]
Small update from my perspective as I used rofi with a bigger history for a bit now  (I set the history size to 75)
1. I see no significant slowdown, though I haven't benchmarked it.
2. Actually the number of items in the history (in the .cache/rofi.... file) is pretty stable somewhere between 40 & 50, with most of the weight in the first 10 items.

This suggests to me that it's not really worthwhile to complicate history handling and that maybe just increasing the hardcoded value a bit (30-40, in my case) would probably be enough. Then again I don't know how well my use case generalizes and what kind of standards you put on your code. Still, I thought this might be interesting data.
